mlrun/package/packagers/default_packager.py

```diff
@@ -0,0 +1,594 @@
+                                      unpickling the object.
+        :param pickle_module_version: The pickle module version. Used to verify the current interpreter module version
+                                      match the one who pickled the object before unpickling it.
+        :param object_module_version: The original object's module version to match to the interpreter's module version.
+
+        :return: The un-pickled python object.
+        """
+        # Get the pkl file to local directory:
+        pickle_path = data_item.local()
+
+        # Add the pickle path to the clearing list:
+        cls.future_clear(path=pickle_path)
+
+        # Unpickle and return:
+        return _Pickler.unpickle(
+            pickle_path=pickle_path,
+            pickle_module_name=pickle_module_name,
+            object_module_name=object_module_name,
+            python_version=python_version,
+            pickle_module_version=pickle_module_version,
+            object_module_version=object_module_version,
+        )
+
+    class _ArgumentsType:
+        """
+        Library class for the arguments type to send for `_validate_method_arguments`. Configurations is the term for
+        the kwargs sent to packing methods and instructions is the term for the kwargs sent in unpacking methods.
+        """
+
+        INSTRUCTIONS = "instructions"
+        CONFIGURATIONS = "configurations"
+
+    @classmethod
+    def _validate_method_arguments(
+        cls, method: MethodType, arguments: dict, arguments_type: str
+    ) -> bool:
+        """
+        Validate keyword arguments to pass to a method. Used for validating log hint configurations for packing methods
+        and instructions for unpacking methods.
+
+        :param method:         The method to validate the arguments for.
+        :param arguments:      Keyword arguments to validate.
+        :param arguments_type: A string to use for the error message. Should be on of the `_ArgumentType` class
+                               variables.
+
+        :return: True if all mandatory arguments are given and False otherwise.
+        """
+        # Get the possible and mandatory (arguments that has no default value) arguments from the functions:
+        possible_arguments = inspect.signature(method).parameters
+        mandatory_arguments = [
+            name
+            for name, parameter in possible_arguments.items()
+            if parameter.default is not inspect._empty
+        ]
+
+        # Validate there are no missing arguments (only mandatory ones):
+        missing_arguments = [
+            mandatory_argument
+            for mandatory_argument in mandatory_arguments
+            if mandatory_argument not in arguments
+        ]
+        if missing_arguments:
+            logger.warn(
+                f"Missing {arguments_type} for {cls.__name__}: {', '.join(missing_arguments)}. The packager won't "
+                f"handle the object."
+            )
```

Comment:
worth adding a call to action for what the user should do? like to add the missing arguments?

Reply:
Great idea! adding it :+1:

Reply:
so you decided to fail instead?

Reply:
it logs a warning message and skip the object as mandatory arguments are missing. the context handler will ignore the exception to not drop the run.